### PR TITLE
MAINT _shared.testing now contains pytest's useful functions

### LINKED ISSUE #2612: Refactor calls to pytest library into utility functions
https://github.com/scikit-image/scikit-image/pull/2468 has been merged, but it would be good to see the `pytest.*` functions getting refactored into a `skimage.util` function instead.

### PR DESCRIPTION
**Pull request by NelleV**
this PR moves pytest's useful function into the testing module to facilitate further changes in testing library.

Ping @stefanv 

\soupault: Closes #2612 .

### CODE REVIEW

File: skimage/io/tests/test_simpleitk.py [hunk 1]
```diff
@@ -55,23 +55,23 @@ def test_bilevel():
 
 """
 #TODO: This test causes a Segmentation fault
```
Comment (soupault) [inline skimage/io/tests/test_simpleitk.py]: LOL!

File: skimage/morphology/tests/test_grey.py [hunk 2]
```diff
@@ -1,15 +1,15 @@
 import os.path
 
 import numpy as np
-from numpy import testing
+from numpy.testing import assert_array_equal, assert_equal
```
Comment (soupault) [inline skimage/morphology/tests/test_grey.py]: Should we wrap these functions as well? And refactor for this file, at least.

File: skimage/util/tests/test_montage.py [hunk 3]
```diff
@@ -1,7 +1,8 @@
 from numpy.testing import assert_equal
 from numpy.testing import assert_array_equal
-import pytest
+from skimage._shared import testing
 
+from skimage._shared import testing
```
Comment (soupault) [inline skimage/util/tests/test_montage.py]: Duplicate row.

File: skimage/morphology/tests/test_grey.py (hunk 2 above)
Comment (NelleV) [inline skimage/morphology/tests/test_grey.py]: @stefanv Do you want to wrap those functions as well?

Comment (soupault) [inline skimage/morphology/tests/test_grey.py]: OK, we can do this later.

**Comment by soupault** [comment]
@NelleV Could you, please, fix the conflict? Is related to the change in https://github.com/scikit-image/scikit-image/pull/2454.

**Comment by soupault** [comment]
An additional `pytest.raises` has been added with the merge of https://github.com/scikit-image/scikit-image/pull/1522 (in `skimage/transform/tests/test_warps.py`).

**Comment by soupault** [comment]
One more test with `pytest.raises`: https://github.com/scikit-image/scikit-image/pull/2631.

**Comment by soupault** [comment]
@scikit-image/core Could we have one more review for this important PR, please?

**Comment by NelleV** [comment]
I'm not sure which tests are broken by me and which tests were already broken… I need to double check everything is fine.

**Comment by stefanv** [comment]
Is this PR stalled or still ongoing?

**Comment by soupault** [comment]
Well, that was an interesting experience :sweat_smile: . I've:
- rebased this PR;
- added `pytest.mark.parameterize` to the wrapper;
- modified the remaining tests to use `skimage._shared.testing` instead of direct imports of `numpy.testing`, `pytest`, and `unittest`;
- fixed low hanging PEP8 issues in the test files.

I think, the PR is pretty much complete for now. @scikit-image/core please, consider reviewing. I understand that the amount of changes is prodigious, yet the changes are relatively simple.

Thanks in advance!

File: skimage/external/test_tifffile.py [hunk 4]
```diff
@@ -68,6 +68,7 @@ def roundtrip(self, dtype, x):
         y = imread(b)
         assert_array_equal(x, y)
 
+<<<<<<< HEAD
```
Comment (stefanv) [inline skimage/external/test_tifffile.py]: All good except for this :)

**Comment by soupault** [comment]
Sorry, there were some issues due to the previous merge/rebase runs.
CI's are green now.

**Comment by stefanv** [comment]
Merging, before this has a chance to become outdated, and block a bunch of other PRs.

Thanks @NelleV and @soupault --- heroic effort here, so many files touched!

**Pull request merged**